File: BackEnd/src/domain/usecases/notifications/DeleteNotificationsUseCase.ts
/**
 * @file DeleteNotificationsUseCase.ts
 * @description Caso de uso para eliminar todas las notificaciones del usuario autenticado.
 * @module Reminders
 * @layer Domain
 *
 * @implements {IDeleteNotificationsUseCase}
 * @injectable
 * @dependencies INotificationRepository
 */

import { injectable, inject } from 'inversify';
import { TYPES } from '../../../core/types.js';
import type { IDeleteNotificationsUseCase } from '../../interfaces/usecases/notifications/IDeleteNotificationsUseCase.js';
import type { INotificationRepository } from '../../repositories/INotificationRepository.js';

/**
 * Elimina todas las notificaciones del usuario.
 *
 * @implements {IDeleteNotificationsUseCase}
 * @injectable
 * @dependencies INotificationRepository
 */
@injectable()
export class DeleteNotificationsUseCase implements IDeleteNotificationsUseCase {
  constructor(
    @inject(TYPES.INotificationRepository) private readonly notifRepo: INotificationRepository,
  ) {}

  /**
   * Elimina notificaciones del usuario autenticado.
   * Si se proporcionan ids, solo elimina esas; si no, elimina todas.
   *
   * @param userId — Id del usuario autenticado.
   * @param ids — Ids de notificaciones específicas (opcional).
   */
  async execute(userId: string, ids?: string[]): Promise<void> {
    if (ids && ids.length > 0) {
      await this.notifRepo.deleteByIds(userId, ids);
    } else {
      await this.notifRepo.deleteAllByUserId(userId);
    }
  }
}

File: FrontEnd/lib/domain/usecases/notifications/delete_notifications_use_case.dart
/// @file delete_notifications_use_case.dart
/// @description Caso de uso para eliminar todas las notificaciones del usuario.
/// @module Reminders
/// @layer Domain
library;

import '../../interfaces/usecases/notifications/i_delete_notifications_use_case.dart';
import '../../repositories/i_notification_repository.dart';

// ═══════════════════════════════════════════════════════════════════════════════
// DELETE NOTIFICATIONS USE CASE
// ═══════════════════════════════════════════════════════════════════════════════

/// Delega en [INotificationRepository.deleteAll].
///
/// [implements] IDeleteNotificationsUseCase
/// [dependencies] INotificationRepository
class DeleteNotificationsUseCase implements IDeleteNotificationsUseCase {
  final INotificationRepository _repository;

  const DeleteNotificationsUseCase({required INotificationRepository repository})
      : _repository = repository;

  @override
  Future<void> execute({List<String>? ids}) => _repository.deleteAll(ids: ids);
}
